icu4c/source/i18n/usearch.cpp

```diff
@@ -572,11 +592,11 @@ inline void initialize(UStringSearch *strsrch, UErrorCode *status)
 */
 static
 void checkBreakBoundary(const UStringSearch *strsrch, int32_t * /*start*/,
-                               int32_t *end)
+                               int32_t *end, UErrorCode *status)
 {
 #if !UCONFIG_NO_BREAK_ITERATION
-    UBreakIterator *breakiterator = strsrch->search->internalBreakIter;
-    if (breakiterator) {
+    UBreakIterator *breakiterator = getBreakIterator(strsrch, status);
+    if (breakiterator != NULL && U_SUCCESS(status)) {
```

Comment:
It looks like it should suffice to check for success but not also for nullptr.
Why did the old code check the pointer? Should it be possible to be nullptr without an error condition? Or just defensive coding against usage of an error-constructed StringSearch object?

Reply:
Good point.
I think it was just defensive coding, as it doesn't appear to be possible to have nullptr without the UErrorCode status also being an error at the same time.
I'll update to only check the error status (and not the pointer) in order to simplify things.
Thanks!